docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+feature will inherit from the abstract class.
+* Specific behavior towards different ```param type```-```parameter``` pairs  will be defined within their 
+own class declarations.
+
+### Architecture
+* The initialisation of ```Ledger``` and ```Entry``` instances can be
+performed with reference to input parameters supplied from the user input.
+* For ledger creation operations, the input from the user is parsed and passed into an initialized ledger instance
+to handle. That is, the handling of input parameters is abstracted out from the tracker classes. 
+<br> The handle operation will set the various attributes within the ledger in accordance to specifications inferred
+from the user input. 
+<br> If the ledger is successfully specified in full, it will be added to a ```ledgerList``` instance within the handler 
+class ```ManualTracker```.
+* For ledger deletion/open, a ledger will need to be selected from the ledger list maintained by the handler class.
+<br>Hence, the input from the user is parsed and passed into a command instance to handle. If the input
+is valid, the ledger list instance will assign a reference to the ledger selected to a public ```currLedger``` 
+attribute. 
+<br>After which, an operation of edit/open would be performed upon the ledger referenced from 
+```currLedger``` in ```ledgerList```.
+
+
+### Logic
+
+#### Input Parsing
+
+##### Input Conventions
+* The user input is composed of the following format:
+```
+    <command> <param type> <parameter> <param type> <parameter> ...
+```
+* The ```command``` string determines the current state of the Finite State Machine, and
+hence the function executed. 
+* The remainder of the string includes a series of  ```param type``` - ```param``` combinations, whereby
+```param type``` indicates the type of the parameter which is to be identified by the user class,
+and ```param``` indicates the parameter that is associated with the ```param type```. 
+
+* Param types are restricted to two types: 
+    * ```/<string>```, requires a corresponding parameter.
+        * Eg. ```param type```: ```/date```
+              <br>  ```param``` : ```2020-04-04```
+    * ```-<string>```, does not require a corresponding parameter. 
+        * Reserved for param types which are used to specify a property to be true/false
+        * Eg. ```-auto```, to specify if an entry has automatic deduction. 
+        
+##### Command Packet
+* A helper class. Contains two particular attributes to store the user input in an organised fashion.
+    * ```commandString``` :  ```String``` Store the command string from the input.
+    * ```paramMap``` : ```HashMap``` Store the pairs of ```param type``` and ```param``` present in the input string.
+        * Key: ```param type```
+        * Value:  ```param```
+
+##### InputParser
+* A helper class. Parses the input string and returns a corresponding ```commandPacket```.
```

Comment:
I'm actually quite confused as to whether commandPacket is suppose to be a property of a class or a class itself, seems like it is a class after reading at the ParamHandler component, but from this line it seems like  a property of some other class.